### DISCUSSION TITLE
LLDPD sending info for all interfaces

### DISCUSSION THREAD

**Discussion by matejzero** [lldpd/lldpd#484] [Q&A]
Hello,

I installed LLDPD on some Dell servers and it is sending info for all interfaces in the server on all ports. For example, on ens1f0, I get info for this port + iDRAC management port and the other port on the network card. I suspect iDRAC is also running LLDP daemon and somehow lldpd sees it as it's neighbour and send it's data too.

Because of that, I get lets of logs on the switches like
```
Local Port 8 receive a mismatch TLV in Auto-negotiation Enable
Local Port 8 receive a mismatch TLV in Link Aggregation Status
```

Is there a way to limit lldpd to only report lldp info for port that it is transmitting out of? I tried using `FILTERING NEIGHBORS` feature, but it doesn't seem to work or maybe I don't know how to properly use it. Anyone had the same issue?

**Comment by vincentbernat** [comment]
LLDP data are multicasted, not directed to a specific neighbor. lldpd will send one LLDPDU on each port. I suppose you use a shared port for iDRAC. You either need to disable lldpd on this port or ensure iDRAC does not transmit a LLDPDU itself.

**Reply by matejzero** [reply]
For iDRAC, I'm using a dedicated port, that is why this is strange. On Lenovo servers, it works OK, but on Dell, it reports all interfaces it can find on the server. Might need to do more debugging.

**Reply by matejzero** [reply]
I think I found the problem. I had Topology LLDP enabled in iDRAC:
> Topology LLDP - In this feature, the LLDP packet goes through all the supported server NIC ports so that an external switch can locate the originating server, NDC port[NIC FQDD], IOM location in the chassis, blade chassis service tag etc. From iDRAC9 4.00.00.00 and later versions, Topology LLDP is available as an option for all PowerEdge servers. The LLDP packets contain server network device connectivity information and are used by I/O modules and external switches to update their configuration. 

If I read this correctly, it will send LLDP packages out of all interfaces, announcing all server interfaces. 

I switched to Discovery LLDP and the result is better:
> Discovery LLDP - In this feature, the LLDP packet goes only through the active iDRAC NIC port in use (dedicated NIC or shared LOM), so an adjacent switch can locate iDRAC connection port in the switch. Discovery LLDP is specific only to the active iDRAC network port and not will not be seen in all the Network ports in the server. Discovery LLDP will have some details of the idrac like IP address, MAC address, service tag etc., so that a switch can automatically discover iDRAC devices connected to it and some data of iDRAC.

**Comment by matejzero** [comment]
I think this is resolved.
